### ISSUE TITLE
More details on the tasks

### ISSUE THREAD

**Issue by jssmith1** [DeveloperLiberationFront/iTrustInterviews#439]
R3:

>- In general, the authors should elaborate a bit more about the tasks. Are they somehow connected? What would be the impact?

>3- Strategies and assumptions are highly dependent on the selected security defects and developers: what is the prior knowledge of the developers regarding the security defects, how frequent are the security defects discussed on the web. The selection of tasks should have been broader to strengthen the results.

R3:
>There are two strong assumptions that make the study resembling a less real world scenario: i) it seems that performance might not be an issue (i.e., Section 3.2.3); ii) it cannot be observed whether the selected strategy in turns introduces a bug.

**Comment by jssmith1** [comment]
Editor:

>My recommendation would be to 1) more deeply elaborate on precisely what participants were asked to do in the task, #439 2) how this might differ from a more authentic task outside of the lab, 3) in the discussion, detail the threats to generalizability that these differences might impose on the results.

**Comment by jssmith1** [comment]
R3 assumptiosn about realisim. To bound the amount of time, we didn't ask participants to assess the quality of their fix.

**Comment by jssmith1** [comment]
- [x] Give more details about tasks. Pull from Flower paper describing iTrust

**Comment by jssmith1** [comment]
What is an authentic task outside the lab? How long? 
Cornell D (2012) Remediation statistics: what does fixing
application vulnerabilities cost? In: RSAConference, San Fransisco,
CA

Also see discussion on what it means to "fix" Doesn't include environment setup, testing, deployment, etc...

|Vulnerability Type | Sample Count | Average Fix (minutes)
|------------------------|---------------|--------------|
Dead Code (unused methods) |465 | 2.6
Poor logging: system output stream | 83 | 2.9
Poor Error Handling: Empty catch block | 180 | 6.8
Lack of Authorization check | 61 | 6.9
Unsafe threading | 301 | 8.5 
ASP.NET non‐serializable object in session | 42 | 9.3
XSS (stored) | 1023 | 9.6
Null Dereference | 157 | 10.2
Missing Null Check | 46 | 15.7
XSS (reflected) | 25 | 16.2
Redundant null check | 21 | 17.1
SQL injection | 30 | 97.5

**Comment by jssmith1** [comment]
Secure development lifecycle...? Maybe not. Talking more about the steps involved in fixing a single vulnerability.

Didn't ask participants to test their proposed fixes. Only focused on the diagnosis phase

**Comment by jssmith1** [comment]
- [x] how do tasks relate to eachother

**Comment by jssmith1** [comment]
We elaborated on what participants were asked to do in the tasks and described in more detail how our tasks relate to tasks in the wild. We already include the verbatim instructions given to participants in the Appendix, but have expanded on the task descriptions in the Methodology section. In particular, we added code experts for each task and some more details about what was involved in each task. To help clarify the difference between our tasks and "more authentic" tasks outside the lab we added the paragraph starting with: "The tasks we chose encompass a subset of the vulnerability mediation activities in the wild."
